### PR TITLE
PAN-OS traffic logs UC implemntation

### PR DESCRIPTION
**Pull request by bakatzir**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/17658

## Description
add support in network traffic check in FW 

## Required version of Demisto
3.0.0 and above

## Does it break backward compatibility?
   - No

## Must have
- [x] Tests
- [x] Documentation - https://github.com/demisto/etc/issues/17762
- [x] Code Review

### CODE REVIEW

File: Integrations/Panorama/Panorama.py [hunk 1]
```diff
@@ -2933,6 +2936,229 @@ def panorama_unregister_ip_tag_command():
...
+    source = demisto.args().get('source')
+    destination = demisto.args().get('destination')
+    receive_time = demisto.args().get('receive_time')
+    application = demisto.args().get('application')
+    to_port = demisto.args().get('to_port')
+    action = demisto.args().get('action')
+
+    if query and (source or destination or receive_time or application or to_port or action):
+        return_error('Use the query argument or the '
+                     'source, destination, receive_time, application, to_port, action arguments to build your query')
```
Comment (yuvalbenshalom) [inline Integrations/Panorama/Panorama.py]: I thin we need a better error message here that explains that you can only use one of them.
Like saying it is mutual exclusive. consult with @kirbles19

File: Integrations/Panorama/Panorama.py [hunk 2]
```diff
@@ -2933,6 +2936,229 @@ def panorama_unregister_ip_tag_command():
...
+
+    result = panorama_query_traffic_logs(number_of_logs, direction, query,
+                                         source, destination, receive_time, application, to_port, action)
+
+    if result['response']['@status'] != 'success':
+        return_error('Query traffic logs failed')
```
Comment (yuvalbenshalom) [inline Integrations/Panorama/Panorama.py]: Panorama is not returning any other information about the failure?

File: Integrations/Panorama/Panorama.py [hunk 3]
```diff
@@ -2933,6 +2936,229 @@ def panorama_unregister_ip_tag_command():
...
+
+    return result
+
+
+def panorama_check_traffic_logs_status_command():
+    job_id = demisto.args().get('job_id')
+    result = panorama_get_traffic_logs(job_id)
+
+    if result['response']['@status'] != 'success':
+        return_error('Query Traffic logs failed')
```
Comment (yuvalbenshalom) [inline Integrations/Panorama/Panorama.py]: same here, can we give the user more info?

File: Integrations/Panorama/Panorama.py [hunk 4]
```diff
@@ -2933,6 +2936,229 @@ def panorama_unregister_ip_tag_command():
...
+                     'source, destination, receive_time, application, to_port, action arguments to build your query')
+
+    result = panorama_query_traffic_logs(number_of_logs, direction, query,
+                                         source, destination, receive_time, application, to_port, action)
+
+    if result['response']['@status'] != 'success':
+        return_error('Query traffic logs failed')
+
+    query_traffic_output = {
+        'JobID': result['response']['result']['job'],
```
Comment (yuvalbenshalom) [inline Integrations/Panorama/Panorama.py]: you need to check keys exist

File: Integrations/Panorama/Panorama.py [hunk 5]
```diff
@@ -2933,6 +2936,229 @@ def panorama_unregister_ip_tag_command():
...
+
+    if result['response']['@status'] != 'success':
+        return_error('Query Traffic logs failed')
+
+    query_traffic_status_output = {
+        'JobID': job_id,
+        'Status': 'Pending'
+    }
+
+    if result['response']['result']['job']['status'] == 'FIN':
```
Comment (yuvalbenshalom) [inline Integrations/Panorama/Panorama.py]: you need to check keys exist

File: Integrations/Panorama/Panorama.py [hunk 6]
```diff
@@ -2933,6 +2936,229 @@ def panorama_unregister_ip_tag_command():
...
+
+    if result['response']['@status'] != 'success':
+        return_error('Query Traffic logs failed')
+
+    query_traffic_logs_output = {
+        'JobID': job_id,
+        'Status': 'Pending'
+    }
+
+    if result['response']['result']['job']['status'] != 'FIN':
```
Comment (yuvalbenshalom) [inline Integrations/Panorama/Panorama.py]: you need to check keys exist

File: Integrations/Panorama/Panorama.py [hunk 7]
```diff
@@ -2933,6 +2936,229 @@ def panorama_unregister_ip_tag_command():
...
+        pretty_traffic_logs_arr.append(pretty_traffic_log)
+    return pretty_traffic_logs_arr
+
+
+def panorama_get_traffic_logs_command():
+    job_id = demisto.args().get('job_id')
+    result = panorama_get_traffic_logs(job_id)
+
+    if result['response']['@status'] != 'success':
+        return_error('Query Traffic logs failed')
```
Comment (yuvalbenshalom) [inline Integrations/Panorama/Panorama.py]: same here, can we give the user more info?

File: Integrations/Panorama/Panorama.py [hunk 8]
```diff
@@ -2933,6 +2936,229 @@ def panorama_unregister_ip_tag_command():
...
+            'ContentsFormat': formats['json'],
+            'Contents': result,
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': tableToMarkdown('Query Traffic Logs status:', query_traffic_logs_output,
+                                             ['JobID', 'Status'], removeNull=True),
+            'EntryContext': {"Panorama.TrafficLogs(val.JobID == obj.JobID)": query_traffic_logs_output}
+        })
+    else:  # FIN
+        query_traffic_logs_output['Status'] = 'Completed'
+        logs = result['response']['result']['log']['logs']
```
Comment (yuvalbenshalom) [inline Integrations/Panorama/Panorama.py]: you need to check keys exist

**Comment by bakatzir** [comment]
error ex:
Screen Shot 2019-06-30 at 12 16 14

**Comment by bakatzir** [comment]
@kirbles19 
https://github.com/demisto/content/blob/a4f0d29ff4f8a3840cbfcea8e1de5fb158980574/Integrations/Panorama/Panorama.py
https://github.com/demisto/content/blob/a4f0d29ff4f8a3840cbfcea8e1de5fb158980574/Integrations/Panorama/Panorama.yml

**Comment by yuvalbenshalom** [comment]
@bakatzir please merge only after Alex's review

File: Integrations/Panorama/Panorama.py (hunk 1 above)
Comment (kirbles19) [reply Integrations/Panorama/Panorama.py]: Which message am I actually supposed to be reviewing?

Comment (bakatzir) [reply Integrations/Panorama/Panorama.py]: @kirbles19  lines 3011-3012.

**Comment by bakatzir** [comment]
@kirbles19 I have forgotten, can you review the new playbook I added as well:
https://github.com/demisto/content/blob/7cd9396da0ebf36356263779721bb5526b2c6404/Playbooks/playbook-PanoramaQueryTrafficLogs.yml

**Pull request merged**